### PR TITLE
Create visual expression builder docs

### PR DESCRIPTION
**Pull request by alexanderahn**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207584391856670

### CODE REVIEW

File: docs/SQL/development/visual-editor/visual-expression-builder/use-the-visual-expression-builder.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+All of the expressions you build using the Visual Expression Builder are converted to code in the Gem and Model Code views. Similarly, edits you make in the Code view, whether it's in an expression or condition editor, will be converted into Visual view.
+
+This allows you to take advantage of both Visual and Code views when building your expressions.
+
+## Build an expression
+
+Using the Visual Expression Builder, you can build an expression using the following modes:
+
+- **Simple Expression**: This mode prompts you to add a target column and an expression.
```
Comment (avcs06) [inline docs/SQL/development/visual-editor/visual-expression-builder/use-the-visual-expression-builder.md:37]: Simple Expression gives the examples but Comparision explains the usage. i think this should be consistent across

`Simple expression`: Allows user to choose an expression, like a column, function, Case statement etc;

`Comparision`: Allows user to compare 2 simple expressions

`Grouping`: Allows user to build complex conditions by allowing the combinations of `Comparision` expressions via `AND` or `OR`

Comment (avcs06) [reply docs/SQL/development/visual-editor/visual-expression-builder/use-the-visual-expression-builder.md:37]: We use `Grouping` in Filter gem, `Comparision` in IF and ELSE IF in Code Table, everywhere else we use the Simple Expression

File: docs/SQL/development/visual-editor/visual-expression-builder/use-the-visual-expression-builder.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    :::tip
+    You can have any number of groups and nestings (a group within a group). And you can change the grouping conditions between `AND` and `OR` by clicking on the labels.
+    :::
+
+## Tips
+
+Here are some additional tips to keep in mind when using the Visual Expression Builder:
+
+- The expression dropdowns support search.
+- Each argument of your function can be another expression since you have the same expression options to choose from.
```
Comment (avcs06) [inline docs/SQL/development/visual-editor/visual-expression-builder/use-the-visual-expression-builder.md:156]: Each argument of your function `is` another expression

File: docs/SQL/development/visual-editor/visual-expression-builder/visual-expression-builder-reference.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+The Visual Expression Builder supports the following expression options:
+
+- **Column**: Allows you to select an input column from your source tables. You can view all of the available input columns from under the dropdown menu or under **Input** on the left-hand side of the Gem dialog.
+- **Value**: Allows you to enter any kind of value.
+  - If you enter a string value, it'll be considered as a string within quotes.
+  - If you enter a number, it'll be considered as a numerical value, but you have the option to click to **Check to read value as string**.
+  - The same applies to a boolean value. For example, if you enter `True`, then it'll be considered a boolean value unless you **Check to read value as string**.
+- **Function**: Includes a list of all of the function category groups and functions that are supported. The list displays each function description, including mandatory arguments.
+- **Conditional**: Allows you to use a conditional `WHEN` clause.
+  - Within `WHEN`, you can select your columns and operator, and within `THEN` you can add a single simple expression.
```
Comment (avcs06) [inline docs/SQL/development/visual-editor/visual-expression-builder/visual-expression-builder-reference.md:28]: In WHEN user can provide a `Comparision expression`. In THEN user can provide a `simple expression`

File: docs/SQL/development/visual-editor/visual-expression-builder/visual-expression-builder-reference.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  - You can add multiple `CASES` of the `WHEN` clause, but you can only have one `ELSE` statement. `ELSE` also uses a simple expression.
+  - You can also add `IF`, `ELSEIF`, or `FOR` conditions between each of your expressions. `FOR` conditions take a variable name and an expression value. `IF` and `ELSEIF` conditions are considered comparisons.
```
Comment (avcs06) [inline docs/SQL/development/visual-editor/visual-expression-builder/visual-expression-builder-reference.md:30]: IF, ELSEIF and FOR are a feature of `Code table` not of Visual expression builder, this is already there in prod we just added a visual counter part to it

Comment (avcs06) [reply docs/SQL/development/visual-editor/visual-expression-builder/visual-expression-builder-reference.md:30]: This is available only in `Expressions` table in Reformat, Aggregate and Join gems

File: docs/SQL/development/visual-editor/visual-expression-builder/use-the-visual-expression-builder.md (hunk 1 above)
Comment (avcs06) [reply docs/SQL/development/visual-editor/visual-expression-builder/use-the-visual-expression-builder.md:37]: Statements I wrote are just for your understanding, please re-write those in correct context you are using

File: docs/SQL/development/visual-editor/visual-expression-builder/visual-expression-builder-reference.md (hunk 3 above)
Comment (alexanderahn) [reply docs/SQL/development/visual-editor/visual-expression-builder/visual-expression-builder-reference.md:28]: Done.

File: docs/SQL/development/visual-editor/visual-expression-builder/visual-expression-builder-reference.md (hunk 4 above)
Comment (alexanderahn) [reply docs/SQL/development/visual-editor/visual-expression-builder/visual-expression-builder-reference.md:30]: Ack. I added a line for the Reformat, Aggregate, and Join Gems.

File: docs/SQL/development/visual-editor/visual-expression-builder/use-the-visual-expression-builder.md (hunk 1 above)
Comment (alexanderahn) [reply docs/SQL/development/visual-editor/visual-expression-builder/use-the-visual-expression-builder.md:37]: Done.

File: docs/SQL/development/visual-editor/visual-expression-builder/use-the-visual-expression-builder.md (hunk 2 above)
Comment (alexanderahn) [reply docs/SQL/development/visual-editor/visual-expression-builder/use-the-visual-expression-builder.md:156]: Done.

**Comment by alexanderahn** [comment]
AvcS provided approval on Slack thread.

**Pull request merged**